调用paddle.layer.img_pool() 时报错

报错如下：
``` python
File "resnet.py", line 69, in resnet_imagenet
    pool1 = paddle.layer.img_pool(input=conv1, pool_size=3, stride=2)
  File "/usr/local/lib/python2.7/dist-packages/paddle/v2/config_base.py", line 52, in wrapped
    out = f(*args, **xargs)
  File "/usr/local/lib/python2.7/dist-packages/paddle/trainer_config_helpers/default_decorators.py", line 53, in __wrapper__
    return func(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/paddle/trainer_config_helpers/layers.py", line 381, in wrapper
    return method(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/paddle/trainer_config_helpers/layers.py", line 2612, in img_pool_layer
    "only (Cudnn)AvgPooling, (Cudnn)MaxPooling are supported"
AssertionError: only (Cudnn)AvgPooling, (Cudnn)MaxPooling are supported
```
该行代码如下
```
    pool1 = paddle.layer.img_pool(input=conv1, pool_size=3, stride=2)
```

--- a/python/paddle/trainer_config_helpers/layers.py
+++ b/python/paddle/trainer_config_helpers/layers.py
@@ -2607,14 +2607,14 @@ def img_pool_layer(input,
         assert input.num_filters is not None
         num_channels = input.num_filters
 
-    assert type(pool_type) in [AvgPooling, MaxPooling, CudnnAvgPooling,
-                               CudnnMaxPooling], \
-        "only (Cudnn)AvgPooling, (Cudnn)MaxPooling are supported"
-
     if pool_type is None:
         pool_type = MaxPooling()
     elif isinstance(pool_type, AvgPooling):
         pool_type.name = 'avg'
+
+    assert type(pool_type) in [AvgPooling, MaxPooling, CudnnAvgPooling,
+                               CudnnMaxPooling], \
+        "only (Cudnn)AvgPooling, (Cudnn)MaxPooling are supported"
 
     type_name = pool_type.name + '-projection' \
         if (

